### PR TITLE
Feat/cli/appmanager/refactoring-commands

### PR DESCRIPTION
**Pull request by eikeland**
## Why
Refactoring of publish commands and helpers.

### CODE REVIEW

File: packages/cli/src/bin/utils/getEndpointUrl.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    if (service || CUSTOM_APPAPI) {
+        return service ?? CUSTOM_APPAPI;
+    }
+
+    /* Env has changed get new api url */
+    if (FUSION_CLI_ENV !== fusionEnv || !process.env.FUSION_CLI_APPAPI) {
+        process.env.FUSION_CLI_ENV = fusionEnv;
+
+        const requestService = await fetch(
+            `https://discovery.ci.fusion-dev.net/service-registry/environments/${fusionEnv}/services/apps`,
```
Comment (odinr) [inline packages/cli/src/bin/utils/getEndpointUrl.ts:29]: use as doc

**Review by odinr — COMMENTED** [review]
fix url to disco 🕺

File: packages/cli/src/bin/utils/getEndpointUrl.ts (hunk 1 above)
Comment (eikeland) [reply packages/cli/src/bin/utils/getEndpointUrl.ts:29]: Done

**Comment by asbjornhaland** [comment]
missing changeset.

**Comment by eikeland** [comment]
> missing changeset.

Its not a release towards main.

File: packages/cli/src/bin/utils/getEndpointUrl.ts (hunk 1 above)
Comment (eikeland) [reply packages/cli/src/bin/utils/getEndpointUrl.ts:29]: adding a todo for now. we need to get it out as alpha and it works. can fix it before beta release.

File: packages/cli/src/bin/publish-application.ts [hunk 2]
```diff
@@ -4,12 +4,15 @@ import { bundleApplication } from './bundle-application.js';
...
+    getEndpointUrl,
+    requireToken,
     tagAppBundle,
-    type FusionEnv,
-} from './utils/app-api.js';
+    uploadAppBundle,
+} from './utils/index.js';
+import type { FusionEnv } from './utils/index.js';
+
+import { exit } from 'node:process';
```
Comment (asbjornhaland) [inline packages/cli/src/bin/publish-application.ts:15]: please move them up to the other imports.

File: packages/cli/src/bin/utils/uploadAppBundle.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        throw new Error(`This app is deleted. HTTP message: ${requestBundle.statusText}`);
+    }
+
+    if (!requestBundle.ok) {
+        throw new Error(
+            `Failed to publish bundle. HTTP status ${requestBundle.status}, ${requestBundle.statusText}`,
+        );
+    }
+
+    return await requestBundle.json();
```
Comment (asbjornhaland) [inline packages/cli/src/bin/utils/uploadAppBundle.ts:55]: Consider switch statement instead.

**Review by asbjornhaland — APPROVED** [review]

**Pull request merged**